bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/handler/FreeAtHomeDeviceHandler.java

```diff
@@ -0,0 +1,587 @@
+                });
+            }
+        } catch (FreeAtHomeHttpCommunicationException e) {
+            logger.debug(
+                    "Communication error during set command {} - at channel {} - full command {} - Error string {}",
+                    deviceID, channelUID.getAsString(), command.toFullString(), e.getMessage());
+
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        FreeAtHomeBridgeHandler freeAtHomeBridge = null;
+
+        Bridge bridge = this.getBridge();
+
+        if (bridge != null) {
+            ThingHandler handler = bridge.getHandler();
+
+            if (handler instanceof FreeAtHomeBridgeHandler bridgeHandler) {
+                freeAtHomeBridge = bridgeHandler;
+            }
+        }
+
+        if (freeAtHomeBridge != null) {
+            updateStatus(ThingStatus.ONLINE);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR);
+            return;
+        }
+
+        FreeAtHomeDatapointGroup dpg = mapChannelUID.get(channelUID);
+
+        // is the datapointgroup invalid
+        if (dpg == null) {
+            logger.debug("Handle command for device (but invalid datapointgroup) {} - at channel {} - full command {}",
+                    deviceID, channelUID.getAsString(), command.toFullString());
+
+            String errInfo = "Datapointgroup is not available in RefreshCommand - channel: " + channelUID.getAsString();
+
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errInfo);
+        } else {
+            if (command instanceof RefreshType) {
+                handleRefreshCommand(freeAtHomeBridge, dpg, channelUID);
+            } else {
+                handleSetCommand(freeAtHomeBridge, dpg, channelUID, command);
+            }
+
+            logger.debug("Handle command for device {} - at channel {} - full command {}", deviceID,
+                    channelUID.getAsString(), command.toFullString());
+        }
+    }
+
+    public void handleEventBasedUpdate(ChannelUID channelUID, State state) {
+        this.updateState(channelUID, state);
+    }
+
+    public void feedbackForVirtualDevice(ChannelUID channelUID, String valueString) {
+        FreeAtHomeBridgeHandler freeAtHomeBridge = null;
+
+        FreeAtHomeDatapointGroup dpg = mapChannelUID.get(channelUID);
+
+        Bridge bridge = this.getBridge();
+
+        if (bridge != null) {
+            ThingHandler handler = bridge.getHandler();
+
+            if (handler instanceof FreeAtHomeBridgeHandler bridgeHandler) {
+                freeAtHomeBridge = bridgeHandler;
+            }
+        }
+
+        if (freeAtHomeBridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "Bride is not valid!");
```

Comment:
Can you add an I18N text here?
```suggestion
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "Bridge is not valid!");
```